src/Sqs/SqsClient.php

```diff
@@ -105,7 +105,68 @@ private function queueUrl()
     }
 
     /**
-     * Validates ReceiveMessage body MD5s
+     * Constructs the expected md5 hash of message attributes according to the encoding
+     * scheme detailed in SQS documentation.
+     *
+     * @param array $message Message containing attributes for validation.
+     *                       Retrieved when using MessageAttributeNames on
+     *                       ReceiveMessage.
+     *
+     * @return string|null The md5 hash of the message attributes according to
+     *                     the encoding scheme. Returns null when there are no
+     *                     attributes.
+     * @link http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-attributes.html#message-attributes-items-validation
+     */
+    public static function getExpectedMessageAttributesMd5($message)
```

Comment:
I'm making a small set of changes along these lines for clarity, thanks for the feedback!